addons/account/models/account_move.py

```diff
@@ -1248,6 +1249,22 @@ def _compute_adjusting_entry_origin_label(self):
             else:
                 move.adjusting_entry_origin_label = False
 
+    @api.depends('move_type')
+    def _compute_allowed_move_types(self):
+        in_move_types = ['in_invoice', 'in_receipt']
+        out_move_types = ['out_invoice', 'out_receipt']
+
+        show_sale_receipts = self.env['ir.config_parameter'].sudo().get_param('account.show_sale_receipts')
+
+        for move in self:
+            default_move_type = move.move_type or move.env.context.get('default_move_type', False)
```

Comment:
```suggestion
            default_move_type = move.move_type or move.env.context.get('default_move_type')
```
I believe we don't need this. It returns `None` instead but it should work the same.